### PR TITLE
Migrate to ActiveMQ Artemis (2-nd attempt)

### PR DESCRIPTION
**Pull request by reta**
Following https://github.com/apache/cxf/pull/919 discussions, restarting the attempt to switch to ActiveMQ Artemis for Jakarta EE branch. 

CC @jimma @ffang @coheigea @rmannibucau

### CODE REVIEW

File: rt/transports/jms/src/test/java/org/apache/cxf/transport/jms/util/MessageListenerTest.java [hunk 1]
```diff
@@ -174,33 +183,21 @@ private void testTransactionalBehaviour(Connection connection, Queue dest) throw
...
     private static Connection createXAConnection(String name, TransactionManager tm) throws JMSException {
-        ActiveMQXAConnectionFactory cf = new ActiveMQXAConnectionFactory("vm://" + name
-                                                                         + "?broker.persistent=false&jms.xaAckMode=1");
-        cf.setRedeliveryPolicy(redeliveryPolicy());
-        XaPooledConnectionFactory cfp = new XaPooledConnectionFactory(cf);
-        cfp.setTransactionManager(tm);
-        cfp.setConnectionFactory(cf);
-        Connection connection = cfp.createConnection();
+        ActiveMQXAConnectionFactory cf = new ActiveMQXAConnectionFactory("vm://" + name);
+        ConnectionFactory cf1 = new ConnectionFactoryProxy(cf, new TransactionHelperImpl(tm));
```
Comment (reta) [inline rt/transports/jms/src/test/java/org/apache/cxf/transport/jms/util/MessageListenerTest.java:194]: @ehsavoie I found a simpler way to integrate with JTA using `ConnectionFactoryProxy` and `TransactionHelperImpl`, the tests are passing right now (except `testConnectionProblemXA`, it has different cause)

Comment (reta) [reply rt/transports/jms/src/test/java/org/apache/cxf/transport/jms/util/MessageListenerTest.java:194]: @ehsavoie awesome! Could you submit a PR against [this branch](https://github.com/reta/cxf/tree/artemis.attempt.2) (it it works for you), thank you!

**Review by jimma — APPROVED** [review]

**Review by ffang — APPROVED** [review]

File: services/wsn/wsn-core/src/main/java/org/apache/cxf/wsn/jms/JmsPublisher.java [hunk 2]
```diff
@@ -128,54 +120,6 @@ protected void validatePublisher(RegisterPublisher registerPublisherRequest)
 
     @Override
     protected void start() throws PublisherRegistrationFailedFault {
-        if (demand) {
-            try {
-                producers = new HashMap<>();
-                advisories = new ArrayList<>();
-                for (TopicExpressionType topic : this.topic) {
```
[the remark this answers is not in the document]
Comment (reta) [reply services/wsn/wsn-core/src/main/java/org/apache/cxf/wsn/jms/JmsPublisher.java:135]: @jimma ActiveMQ's advisories have been re-implemented using ActiveMQ Artemis management notifications :-) I think we are good to go with this one

[the remark this answers is not in the document]
Comment (jimma) [reply services/wsn/wsn-core/src/main/java/org/apache/cxf/wsn/jms/JmsPublisher.java:135]: @reta  This is great. Thanks for the effort !

**Pull request merged**